src/language-js/print/flow.js

```diff
@@ -98,6 +99,10 @@ function printFlow(path, options, print) {
       return printIntersectionType(path, options, print);
     case "UnionTypeAnnotation":
       return printUnionType(path, options, print);
+    case "ConditionalType":
+      return printTernary(path, options, print);
+    case "InferType":
```

Comment:
Though it's one line, but please share logic with `TSInferType` extract `printInferType`.